### PR TITLE
Nodejs cx wrapper

### PR DESCRIPTION
**Pull request by RafalSumislawski**
The main goal of this PR is to solve issues with instrumenting some versions of code emitted by bundlers like webpack / esbuild. With some configurations these bundlers emit a handler function definiton that cannot be redefined by the usual instrumentation mechanism of opentelemetry-js. The solution is to instead:
*  direct AWS to our handler function (see redefinition of `_HANDLER` in `otel-handler`) 
* and load and call the user handler function, wrapped, from our handler (see `index.ts` `handler`)
This way no redefinition is needed, so it works with various outputs from bundlers.

This new wrapper has been created by moving the `wrapper.ts` into a new `cx-wrapper` module (I needed it to be a module, not a top level script). Using `UserFunction.js` which is provided in the lambda runtime, so we use AWS's code, this gives us guarantee that our function loading works the same as AWS's. Adding a `handler` function. Additionally I've split the `wrapper.ts` into multiple files divided by their responsibility (I initially did that because I thought we will need more than one version of the wrapper, reusing the rest of the code. In the end we have just one wrapper, but the split into files still makes sense).

### CODE REVIEW

File: ci-scripts/build_nodejs_layer.sh [hunk 1]
```diff
@@ -22,6 +22,15 @@ fi
...
 IITM_PATH=$(realpath $IITM_PATH)
 
+CWD=$(pwd)
+
+echo OPENTELEMETRY_JS_CONTRIB_PATH=$OPENTELEMETRY_JS_CONTRIB_PATH
+echo OPENTELEMETRY_JS_PATH=$OPENTELEMETRY_JS_PATH
+echo IITM_PATH=$IITM_PATH
+echo CWD=$CWD
+
+npm cache clean --force
```
Comment (rtimush) [inline ci-scripts/build_nodejs_layer.sh]: It would be beneficial to add a comment why is this necessary.

**Review by rtimush — APPROVED** [review]

File: ci-scripts/build_nodejs_layer.sh (hunk 1 above)
Comment (RafalSumislawski) [reply ci-scripts/build_nodejs_layer.sh]: It probably isn't. I will remove it. 

This is crazy but the only way to reliably build this all is to do fresh checkout of all repositories, because none of the builds can do a correct incremental build taking into account a removal of a source file. And I added this line while fighting this issue, but of course that doesn't help.
